src/wp-includes/template.php

```diff
@@ -965,50 +965,153 @@ function wp_finalize_template_enhancement_output_buffer( string $output, int $ph
 
 	$filtered_output = $output;
 
-	/**
-	 * Filters the template enhancement output buffer prior to sending to the client.
-	 *
-	 * This filter only applies the HTML output of an included template. This filter is a progressive enhancement
-	 * intended for applications such as optimizing markup to improve frontend page load performance. Sites must not
-	 * depend on this filter applying since they may opt to stream the responses instead. Callbacks for this filter are
-	 * highly discouraged from using regular expressions to do any kind of replacement on the output. Use the HTML API
-	 * (either `WP_HTML_Tag_Processor` or `WP_HTML_Processor`), or else use {@see DOM\HtmlDocument} as of PHP 8.4 which
-	 * fully supports HTML5.
-	 *
-	 * Important: Because this filter is applied inside an output buffer callback (i.e. display handler), any callbacks
-	 * added to the filter must not attempt to start their own output buffers. Otherwise, PHP will raise a fatal error:
-	 * "Cannot use output buffering in output buffering display handlers."
-	 *
-	 * @since 6.9.0
-	 *
-	 * @param string $filtered_output HTML template enhancement output buffer.
-	 * @param string $output          Original HTML template output buffer.
-	 */
-	$filtered_output = (string) apply_filters( 'wp_template_enhancement_output_buffer', $filtered_output, $output );
+	$did_just_catch = false;
+
+	$error_log = array();
+	set_error_handler(
+		static function ( int $level, string $message, ?string $file = null, ?int $line = null ) use ( &$error_log, &$did_just_catch ) {
+			// Switch a user error to an exception so that it can be caught and the buffer can be returned.
+			if ( E_USER_ERROR === $level ) {
+				throw new Exception( __( 'User error triggered:' ) . ' ' . $message );
+			}
+
+			// Display a caught exception as an error since it prevents any of the output buffer filters from applying.
+			if ( $did_just_catch ) { // @phpstan-ignore if.alwaysFalse (The variable is set in the catch block below.)
+				$level = E_USER_ERROR;
+			}
+
+			// Capture a reported error to be displayed by appending to the processed output buffer if display_errors is enabled.
+			if ( error_reporting() & $level ) {
+				$error_log[] = compact( 'level', 'message', 'file', 'line' );
+			}
+			return false;
+		}
+	);
+	$original_display_errors = ini_get( 'display_errors' );
+	if ( $original_display_errors ) {
+		ini_set( 'display_errors', 0 );
```

Comment:
Not sure if this nit matters, so asking here before creating a new ticket +PR, but officially for PHP <8.1, the second param should be a string: https://www.php.net/manual/en/function.ini-set.php

<img width="1603" height="231" alt="image" src="https://github.com/user-attachments/assets/f89b44cc-e506-46f1-afd9-cbd469a10480" />


https://phpstan.org/r/371b5bf7-6e70-45db-9d94-919d1ae49a5e